src/status/core.nim

```diff
@@ -1,16 +1,30 @@
+import json
 import libstatus
-import chat
 import nimcrypto
-
-proc startMessenger*() =
-  chat.startMessenger()
+import utils
 
 proc callRPC*(inputJSON: string): string =
   return $libstatus.callRPC(inputJSON)
 
-proc callPrivateRPC*(inputJSON: string): string =
+proc callPrivateRPCRaw*(inputJSON: string): string =
   return $libstatus.callPrivateRPC(inputJSON)
 
+proc callPrivateRPC*(methodName: string, payload = %* []): string =
+  try:
+    let inputJSON = %* {
+      "jsonrpc": "2.0",
+      "method": methodName,
+      "params": %payload
+    }
+    echo inputJSON
+    result = $libstatus.callPrivateRPC($inputJSON)
+  except:
+    echo "error doing rpc request"
```

Comment:
Use nim-chronicles   `error "Error doing RPC request", method=methodName payload=payload` ,

Reply:
BTW, Chronicles has a short-cut when the property name matches a variable name. It's ok to say:

```nim
error "Error doing RPC request", method=methodName, payload
```